### PR TITLE
Fix part #17,#382: Hi fi topic audio bar

### PR DESCRIPTION
**Pull request by nikitamarysolomanpvt**
## Explanation
Hi fi topic audio bar 

## High-fi Checklist
- [x] Seekbar and its animation
Screenshot 2019-11-08 at 3 24 11 PM

- [x] AudioBar along with Volume icon in toolbar
Screenshot 2019-11-08 at 3 24 37 PM

## Issue Link
https://github.com/oppia/oppia-android/issues/17

## Mock Link
[https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/9ae3d04d-9845-4b75-abf8-8f69e8823fb1/PM-IN-AU-Playing-2-/](https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/9ae3d04d-9845-4b75-abf8-8f69e8823fb1/PM-IN-AU-Playing-2-/)

## GIF 

new

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by veena14cs** [comment]
In Pixel 3a Xl device gap is too much between play button and seekbar and also between language. Please check in the screenshot.

Screenshot_20191121-175159

**Comment by veena14cs** [comment]
@mschanteltc can we have lighter background for audio bar or seekbar and thumb color lighter then the background. Because seekbar is not clearly visible on audiobar.

**Comment by nikitamarysolomanpvt** [comment]
> In Pixel 3a Xl device gap is too much between play button and seekbar and also between language. Please check in the screenshot.
> 
> Screenshot_20191121-175159

Done..

**Review by rt4914 — APPROVED** [review]
This looks good and discussed with @nikitamarysolomanpvt earlier, the audiobar does have issues, which will be picked up once we get clarifications from Chantel.

**Comment by mschanteltc** [comment]
I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3. 
Audio Scrub

**Comment by nikitamarysolomanpvt** [comment]
> I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.

Please see the drop-shadow 
opacity:54%
small
drop-shadow opacity:20%
small-2
Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...

**Comment by mschanteltc** [comment]
> > I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.
> 
> Please see the drop-shadow
> opacity:54%
> small
> drop-shadow opacity:20%
> small-2
> Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...

I think the audio thumb with the 54% opacity is looking great! It's more evident that it appears above the scrub and has a "tappable" appearance.

**Comment by nikitamarysolomanpvt** [comment]
> > > I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.
> > 
> > 
> > Please see the drop-shadow
> > opacity:54%
> > small
> > drop-shadow opacity:20%
> > small-2
> > Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...
> 
> I think the audio thumb with the 54% opacity is looking great! It's more evident that it appears above the scrub and has a "tappable" appearance.

Ok then iam not using the svg image that i requested you.

**Comment by mschanteltc** [comment]
> > > > I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.
> > > 
> > > 
> > > Please see the drop-shadow
> > > opacity:54%
> > > small
> > > drop-shadow opacity:20%
> > > small-2
> > > Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...
> > 
> > 
> > I think the audio thumb with the 54% opacity is looking great! It's more evident that it appears above the scrub and has a "tappable" appearance.
> 
> Ok then iam not using the svg image that i requested you.

LGTM

**Review by veena14cs — APPROVED** [review]
LGTM Please check if it passes accessibility test

File: app/src/main/res/drawable/audio_seekbar_background.xml [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+
+      <corners android:radius="7dp" />
+    </shape>
+  </item>
+  <item android:id="@android:id/progress">
+    <clip>
+      <shape>
+        <solid
+          android:color="@color/colorAccent"/>
+        <corners android:radius="7dp" />
```
Comment (BenHenning) [inline app/src/main/res/drawable/audio_seekbar_background.xml]: Shouldn’t all dimensions be in multiples of 4dp per material guidelines?

**Review by BenHenning — APPROVED** [review]
Thanks!

File: app/src/main/res/drawable/audio_seekbar_background.xml (hunk 1 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/drawable/audio_seekbar_background.xml]: Done

**Comment by nikitamarysolomanpvt** [comment]
> > > > > I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.
> > > > 
> > > > 
> > > > Please see the drop-shadow
> > > > opacity:54%
> > > > small
> > > > drop-shadow opacity:20%
> > > > small-2
> > > > Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...
> > > 
> > > 
> > > I think the audio thumb with the 54% opacity is looking great! It's more evident that it appears above the scrub and has a "tappable" appearance.
> > 
> > 
> > Ok then iam not using the svg image that i requested you.
> 
> LGTM

## Accessibility Scanner Results: Oppia - 2019-11-25-11:25:39

screenshot_Oppia_2019-11-25-11_25_39Item label
org.oppia.app:id/sbAudioProgress
This item may not have a label readable by screen readers.

Touch target
org.oppia.app:id/tvAudioLanguage

@mschanteltc Could you please suggest ...
This item's width is 46dp. Consider making the width of this touch target 48dp or larger.

**Comment by nikitamarysolomanpvt** [comment]
> LGTM Please check if it passes accessibility test

Done

**Comment by nikitamarysolomanpvt** [comment]
@rt4914 & @veena14cs i fixed #382 so PTAL

**Comment by rt4914** [comment]
@nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.

File: app/src/main/res/drawable/audio_seekbar_background.xml [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
-  <stroke
-    android:width="4dp"
-    android:color="@color/white" />
-</shape>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+  <item android:id="@android:id/background">
+    <shape>
+      <solid
+        android:color="@color/white"/>
+
```
Comment (veena14cs) [inline app/src/main/res/drawable/audio_seekbar_background.xml]: Remove this extra space.

**Review by veena14cs — APPROVED** [review]
LGTM one nit change.

**Review by rt4914 — APPROVED** [review]
LGTM

**Comment by nikitamarysolomanpvt** [comment]
> @nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.

@rt4914 Which topic you tried. Please be specific when you mention such issues... is this issue there for all topics or in specific topic?

**Comment by rt4914** [comment]
> > @nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.
> 
> @rt4914 Which topic you tried. Please be specific when you mention such issues... is this issue there for all topics or in specific topic?

I checked on all topics and it was not working. But on giving a second look actually it was working on Real device and it was failing on Emulator. That's why I approved it in second pass.

**Comment by nikitamarysolomanpvt** [comment]
> > > @nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.
> > 
> > 
> > @rt4914 Which topic you tried. Please be specific when you mention such issues... is this issue there for all topics or in specific topic?
> 
> I checked on all topics and it was not working. But on giving a second look actually it was working on Real device and it was failing on Emulator. That's why I approved it in second pass.

@rt4914 I checked and it's working perfectly on both devices and emulator... PTAL

**Comment by rt4914** [comment]
> > > > @nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.
> > > 
> > > 
> > > @rt4914 Which topic you tried. Please be specific when you mention such issues... is this issue there for all topics or in specific topic?
> > 
> > 
> > I checked on all topics and it was not working. But on giving a second look actually it was working on Real device and it was failing on Emulator. That's why I approved it in second pass.
> 
> @rt4914 I checked and it's working perfectly on both devices and emulator... PTAL

On further investigation it was found that it is failing on Android API 19.

**Comment by rt4914** [comment]
> > > > > @nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.
> > > > 
> > > > 
> > > > @rt4914 Which topic you tried. Please be specific when you mention such issues... is this issue there for all topics or in specific topic?
> > > 
> > > 
> > > I checked on all topics and it was not working. But on giving a second look actually it was working on Real device and it was failing on Emulator. That's why I approved it in second pass.
> > 
> > 
> > @rt4914 I checked and it's working perfectly on both devices and emulator... PTAL
> 
> On further investigation it was found that it is failing on Android API 19.

I have filed this as an issue #475

**Comment by mschanteltc** [comment]
> > > > > > I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.
> > > > > 
> > > > > 
> > > > > Please see the drop-shadow
> > > > > opacity:54%
> > > > > small
> > > > > drop-shadow opacity:20%
> > > > > small-2
> > > > > Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...
> > > > 
> > > > 
> > > > I think the audio thumb with the 54% opacity is looking great! It's more evident that it appears above the scrub and has a "tappable" appearance.
> > > 
> > > 
> > > Ok then iam not using the svg image that i requested you.
> > 
> > 
> > LGTM
> 
> ## Accessibility Scanner Results: Oppia - 2019-11-25-11:25:39
> screenshot_Oppia_2019-11-25-11_25_39Item label
> org.oppia.app:id/sbAudioProgress
> This item may not have a label readable by screen readers.
> 
> Touch target
> org.oppia.app:id/tvAudioLanguage
> 
> @mschanteltc Could you please suggest ...
> This item's width is 46dp. Consider making the width of this touch target 48dp or larger.

Hi Nikita,

I fixed the language selector button to be 48 px in width. LMK what you think.

**Comment by nikitamarysolomanpvt** [comment]
Thats already done. I set it as 48dp

**Pull request merged**